src/plugins/data/common/kbn_field_types/index.ts

```diff
@@ -6,11 +6,39 @@
  * Side Public License, v 1.
  */
 
-export { KbnFieldType } from './kbn_field_type';
+// NOTE: trick to mark exports as deprecated (only for constants and types, but not for interfaces, classes or enums)
+import {
+  castEsToKbnFieldTypeName as oldCastEsToKbnFieldTypeName,
+  getFilterableKbnTypeNames as oldGetFilterableKbnTypeNames,
+  getKbnFieldType as oldGetKbnFieldType,
+  getKbnTypeNames as oldGetKbnTypeNames,
+  KbnFieldType,
+} from '@kbn/field-types';
+
+/**
+ * @deprecated import from "@kbn/field-types" instead
```

Comment:
I think it could be nice if we use identical messages across Kibana. 
Maybe align with the format I started (or change to yours if you prefer).

Reply:
I agree, I made this message a bit nicer and add `@removeBy` tag. 

We need to add a `@removeBy` tag and change `kbn/es-query` -> `@kbn/es-query` in other messages, but I don't want to do this in this pr. I would prefer a separate pr instead. There we can also align those.

Reply:
I made https://github.com/elastic/kibana/pull/107175
Mind reviewing?